src/monexamine.cpp

```diff
@@ -380,11 +394,18 @@ void monexamine::attach_or_remove_saddle( monster &z )
 {
     if( z.has_effect( effect_saddled ) ) {
         z.remove_effect( effect_saddled );
-        item riding_saddle( "riding_saddle", 0 );
-        g->u.i_add( riding_saddle );
+        g->u.i_add( *z.tack_item );
+        z.tack_item = cata::nullopt;
     } else {
+        int pos = tack_pos();
+
+        if( pos == INT_MIN ) {
+            add_msg( _( "Never mind." ) );
+            return;
+        }
         z.add_effect( effect_saddled, 1_turns, num_bp, true );
-        g->u.use_amount( "riding_saddle", 1 );
+        z.tack_item = g->u.i_at( pos );
+        g->u.use_amount( ( *z.tack_item ).typeId(), 1 );
```

Comment:
I suggest changing `tack_pos` to return the `item_location` directly. Than you can check it here for validness (instead of `pos == INT_MIN`). After adding the item to the monster, it can remove from the player via a function of `item_location` (`remove_item`).